be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -149,17 +163,36 @@ Status HdfsTextScanner::_parse_csv(int chunk_size, ChunkPtr* chunk) {
         }
 
         bool has_error = false;
-        for (int j = 0, k = 0; j < _scanner_params.materialize_slots.size(); j++) {
+        int num_materialize_columns = _scanner_params.materialize_slots.size();
+        for (int j = 0; j < num_materialize_columns; j++) {
+            int index = _scanner_params.materialize_index_in_chunk[j];
             const Slice& field = fields[_scanner_params.materialize_slots[j]->id() - 1];
```

Comment:
Can you be sure that index of `fields` is `materialize_slots[j]->id()-1`?

Reply:
@dirtysalt i think for text, if we do not store original column id int external table metadata. it is difficult to align.
result below is impala shows:
columns of test data is :
CountryID int , CountryName string, Capital string, Population string
```
create external table if not exists countries_list_1(Capital string, CountryID int ,  CountryName string, Population string) comment 'List of countries' row format delimited fields terminated by ',' stored as textfile location '/tmp/yuzhou';
```
```
[emr-header-1.cluster-259859:21000] default> select * from countries_list_1;
Query: select * from countries_list_1
Query submitted at: 2022-01-15 10:56:29 (Coordinator: http://emr-worker-3.cluster-259859:25000)
Query progress can be monitored at: http://emr-worker-3.cluster-259859:25000/query_plan?query_id=e5401f6bd049906f:8e80a59e00000000
+-----------+-----------+-------------+------------+
| capital   | countryid | countryname | population |
+-----------+-----------+-------------+------------+
| countryid | NULL      | capital     | population |
| 0         | NULL      | BeiJing     | 80         |
| 1         | NULL      | Wash        | 328        |
| 0         | NULL      | BeiJing     | 80         |
| 1         | NULL      | Wash        | 328        |
+-----------+-----------+-------------+------------+
WARNINGS: Error converting column: 1 to INT
Error parsing row: file: hdfs://emr-header-1.cluster-259859:9000/tmp/yuzhou/countries_1.csv, before offset: 75
Error converting column: 1 to INT
Error parsing row: file: hdfs://emr-header-1.cluster-259859:9000/tmp/yuzhou/countries_1.csv, before offset: 75
Error converting column: 1 to INT
Error parsing row: file: hdfs://emr-header-1.cluster-259859:9000/tmp/yuzhou/countries_1.csv, before offset: 75
Error converting column: 1 to INT
Error parsing row: file: hdfs://emr-header-1.cluster-259859:9000/tmp/yuzhou/countries.csv, before offset: 34
Error converting column: 1 to INT
Error parsing row: file: hdfs://emr-header-1.cluster-259859:9000/tmp/yuzhou/countries.csv, before offset: 34
```

Reply:
Got it.